MaxText/pyconfig.py

```diff
@@ -87,6 +87,8 @@ def user_init(raw_keys):
     run_name = raw_keys["run_name"]
     assert run_name, "Erroring out, need a real run_name"
     base_output_directory = raw_keys["base_output_directory"]
+    assert base_output_directory, "Erroring out, need a real base_output_directory"
+    assert raw_keys['dataset_path'], "Erroring out, need a real dataset_path"
```

Comment:
I think it would be nice to also include a link to the readme section for dowloading the dataset here, something like "Erroring out, need a real dataset_path. See  instructions for dowloading the c4 dataset here: https://github.com/google/maxtext/blob/main/README.md#getting-started-download-dataset-and-configure."